### PR TITLE
[SDXL, Docs] Textual inversion

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

Adds better docs on how to use textual inversion with diffusers. Also cc @apolinario

### CODE REVIEW

File: docs/source/en/using-diffusers/sdxl.md [hunk 1]
```diff
@@ -397,6 +397,8 @@ image = pipeline(prompt=prompt, prompt_2=prompt_2).images[0]
     <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/sdxl-double-prompt.png" alt="generated image of an astronaut in a jungle in the style of a van gogh painting"/>
 </div>
 
+SDXL can be enhanced with textual inversion embeddings, in a similar way to Stable Diffusion 1 and 2 as explained [here](https://huggingface.co/docs/diffusers/v0.21.0/en/using-diffusers/textual_inversion_inference#textual-inversion)
```
Comment (patrickvonplaten) [inline docs/source/en/using-diffusers/sdxl.md:400]:
```suggestion
SDXL can be enhanced with textual inversion embeddings, in a similar way to Stable Diffusion 1 and 2 as explained [here](./textual_inversion_inference#stable-diffusion-xl)
```

File: docs/source/en/using-diffusers/textual_inversion_inference.md [hunk 2]
```diff
@@ -69,3 +71,52 @@ grid
 <div class="flex justify-center">
     <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/textual_inversion_inference.png">
 </div>
+
+
+## Stable Diffusion XL
+
+Stable Diffusion XL can also use textual inversion vectors for inference. In contrast to Stable Diffusion 1 or 2, Stable Diffusion XL has two text encoders. Therefore textual inversion embeddings for SDXL tend to also have two embedding vectors - one for each text encoder model.
```
Comment (stevhliu) [inline docs/source/en/using-diffusers/textual_inversion_inference.md:78]:
```suggestion
Stable Diffusion XL (SDXL) can also use textual inversion vectors for inference. In contrast to Stable Diffusion 1 and 2, SDXL has two text encoders so you'll need two textual inversion embeddings - one for each text encoder model.
```

File: docs/source/en/using-diffusers/textual_inversion_inference.md [hunk 3]
```diff
@@ -69,3 +71,52 @@ grid
...
+
+In order to use a textual inversion embedding for Stable Diffusion XL, first make sure to download the file 
+and load it as a dict to have a look at the structure.
```
Comment (stevhliu) [inline docs/source/en/using-diffusers/textual_inversion_inference.md:81]:
```suggestion
Let's download the SDXL textual inversion embeddings and have a closer look at it's structure:
```

File: docs/source/en/using-diffusers/textual_inversion_inference.md [hunk 4]
```diff
@@ -69,3 +71,52 @@ grid
...
+```py
+from huggingface_hub import hf_hub_download
+from safetensors.torch import load_file
+
+file = hf_hub_download("dn118/unaestheticXL", filename="unaestheticXLv31.safetensors")
+state_dict = load_file(file)
+state_dict
+```
+
+**Output**:
```
Comment (stevhliu) [inline docs/source/en/using-diffusers/textual_inversion_inference.md:92]:
```suggestion

```

File: docs/source/en/using-diffusers/textual_inversion_inference.md [hunk 5]
```diff
@@ -69,3 +71,52 @@ grid
...
+         ...,
+         [-0.0170,  0.0213,  0.0143,  ..., -0.0302, -0.0240, -0.0362]],
+ 'clip_l': tensor([[ 0.0023,  0.0192,  0.0213,  ..., -0.0385,  0.0048, -0.0011],
+         ...,
+         [ 0.0475, -0.0508, -0.0145,  ...,  0.0070, -0.0089, -0.0163]],
+```
+
+As we can see we have indeed two tensors, one for `"clip-g"` and one for `"clip-l"`.
+`"clip-g"` corresponds to the bigger text encoder in Stable Diffusion XL and refers to 
+`pipe.text_encoder_2` whereas `"clip-l"` refers to `pipe.text_encoder`.
```
Comment (stevhliu) [inline docs/source/en/using-diffusers/textual_inversion_inference.md:104]:
```suggestion
There are two tensors, `"clip-g"` and `"clip-l"`.
`"clip-g"` corresponds to the bigger text encoder in SDXL and refers to 
`pipe.text_encoder_2` and `"clip-l"` refers to `pipe.text_encoder`.
```

File: docs/source/en/using-diffusers/textual_inversion_inference.md [hunk 6]
```diff
@@ -69,3 +71,52 @@ grid
...
+
+Now we can load each tensors separately by passing the correct text encoder and tokenizer
```
Comment (stevhliu) [inline docs/source/en/using-diffusers/textual_inversion_inference.md:106]:
```suggestion
Now you can load each tensor separately by passing them along with the correct text encoder and tokenizer
```

File: docs/source/en/using-diffusers/sdxl.md [hunk 7]
```diff
@@ -397,6 +397,8 @@ image = pipeline(prompt=prompt, prompt_2=prompt_2).images[0]
     <img src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/diffusers/sdxl-double-prompt.png" alt="generated image of an astronaut in a jungle in the style of a van gogh painting"/>
 </div>
 
+SDXL can be enhanced with textual inversion embeddings, in a similar way to Stable Diffusion 1 and 2 as explained [here](textual_inversion_inference#stable-diffusion-xl)
```
Comment (stevhliu) [inline docs/source/en/using-diffusers/sdxl.md:400]: For a smoother transition :)

```suggestion
The dual text-encoders also support textual inversion embeddings that need to be loaded separately as explained in the [SDXL textual inversion](textual_inversion_inference#stable-diffusion-xl] section.
```

**Review by stevhliu — APPROVED** [review]
Thanks for adding! 🙂

**Review by sayakpaul — APPROVED** [review]
Sweet!

**Pull request merged**